### PR TITLE
Add explicit move assignment operator for UiaTuple

### LINKED ISSUE #66: Cannot use move assignment operator for UiaTuple
There is no explicit reason to avoid UiaTuple make move assignments.
But UiaTuple cannot just get the inherent function from UiaTypeBase, so if we want to make a move assignment for UiaTuple, we may need to define the move assignment operator and move constructor for UiaTuple explicitly.

### PR DESCRIPTION
**Pull request by pengyuanjin**
There is no explicit reason to avoid UiaTuple make move assignments.
But UiaTuple cannot just get the inherent function from UiaTypeBase.
This PR defines the move assignment operator and move constructor for UiaTuple explicitly.

Closes #66

### CODE REVIEW

File: src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.h [hunk 1]
```diff
@@ -1907,6 +1907,9 @@ namespace UiaOperationAbstraction
             return false;
         }
 
+        UiaTuple(UiaTuple&&) = default;
```
Comment (beweedon) [inline src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.h:1910]: [Rule of Five](https://en.wikipedia.org/wiki/Rule_of_three_(C%2B%2B_programming)#Rule_of_Five), you should probably define a copy assignment operator and destructor as well. Also might be good to put them all next to each other.

Comment (pengyuanjin) [reply src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.h:1910]: I noticed that `UiaTypeBase` doesn't have a destructor but has the other four. Shall we add that?

File: src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.h [hunk 2]
```diff
@@ -1891,6 +1891,12 @@ namespace UiaOperationAbstraction
         }
 
         UiaTuple(const UiaTuple&) = default;
+        UiaTuple& operator=(const UiaTuple&) = delete;
```
Comment (csauls) [inline src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.h:1894]: Why is this deleted but the companion method is default? In general, either both are allowed or both are deleted. I think this creates confusion.

Comment (pengyuanjin) [reply src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.h:1894]: I am doing this because `UiaTypeBase` is doing similar:
```
        UiaTypeBase(const UiaTypeBase&) = default;
        UiaTypeBase& operator=(const UiaTypeBase&) = delete;
```

**Review by zhayuli — APPROVED** [review]

File: src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.h (hunk 1 above)
Comment (beweedon) [reply src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.h]: Hmm, yeah let's add that as part of this PR.

**Review by csauls — APPROVED** [review]

**Comment by beweedon** [comment]
@pengyuanjin I'll approve once we add the destructor to UiaTypeBase

**Comment by pengyuanjin** [comment]
@beweedon Added.

**Review by beweedon — APPROVED** [review]
